### PR TITLE
Phase 2

### PR DESCRIPTION
**Pull request by jnav24**

### CODE REVIEW

**Comment by jnav24** [comment]
@A5hleyRich

File: resources/js/components/BalanceList.vue [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      Expense,
+      Paginate,
+      Summary,
+      Nav,
+    },
+
+    created() {
+      this.expenseData = JSON.parse(JSON.stringify(this.expenses));
+
+      this.$bus.$on(UPDATE_ENTRY, data => {
```
Comment (gilbitron) [inline resources/js/components/BalanceList.vue:46]: These callbacks could probably be moved to component methods to clean things up a bit.

File: resources/js/components/ImportCSVModal.vue [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          }
+        }
+
+        return null;
+      },
+    },
+
+    data() {
+      return {
+        acceptedTypes: ['text/csv'],
```
Comment (gilbitron) [inline resources/js/components/ImportCSVModal.vue:58]: This should be a computed prop as it isn't going to change.

File: resources/js/components/ImportCSVModal.vue [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    components: {
+      Label,
+      Link,
+      Modal,
+      SaveActions,
+    },
+
+    computed: {
+      canSubmit: function () {
+        return !this.loading && this.file && this.acceptedTypes.includes(this.file.type);
```
Comment (gilbitron) [inline resources/js/components/ImportCSVModal.vue:42]: If I refreshed the page `this.loading` would be false but the process would still be happening in the background.

File: app/Http/Controllers/ExpenseController.php [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        ]);
+    }
+
+    public function import(Request $request)
+    {
+        $validate = $request->validate([
+            'file' => 'required|file|mimes:csv,txt',
+        ]);
+
+        Excel::import(new ExpenseImport, $validate['file']);
```
Comment (gilbitron) [inline app/Http/Controllers/ExpenseController.php:128]: This should probably be run in a queue job to accommodate for processing massive files.

File: app/Http/Controllers/ExpenseController.php [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        $expense->delete();
+    }
+
+    public function getCsvRowCount(Request $request)
+    {
+        $validate = $request->validate([
+            'file' => 'required|file|mimes:csv,txt',
+        ]);
+
+        $count = (new ExpenseImport)->toCollection($validate['file'])->shift()->count();
```
Comment (gilbitron) [inline app/Http/Controllers/ExpenseController.php:115]: If the file was huge converting it to a collection like this might cause memory issues.

**Review by gilbitron — COMMENTED** [review]
@jnav24 Thanks for submitting your phase 2 code. I've left some feedback in the comments below.

**Pull request merged**